app/controllers/new-signup.js

```diff
@@ -0,0 +1,76 @@
+import Controller from '@ember/controller';
+import { action, set } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import registerUser from '../utils/register-api';
+
+const SIGNUP_STEPS = [
+  'get-started',
+  'firstName',
+  'lastName',
+  'username',
+  'thank-you',
+];
+
+export default class SignupController extends Controller {
+  @tracked isSubmitClicked = false;
+  @tracked isButtonDisabled = true;
+
+  @tracked currentStepIndex = 0;
+
+  @tracked dev = false;
+  @tracked userDetails = {
+    firstName: '',
+    lastName: '',
+    username: '',
+  };
+  @tracked errorMessage;
+
+  get currentStep() {
+    return SIGNUP_STEPS[this.currentStepIndex];
+  }
+
+  @action changeRouteParams() {
+    const nextStepIndex = this.currentStepIndex + 1;
+
+    console.log(nextStepIndex, SIGNUP_STEPS.length);
+
+    if (nextStepIndex >= SIGNUP_STEPS.length) {
```

Comment:
How do we ensure that an error in `nextStepIndex` does not erroneously trigger the `signup()`